app/api/shorten.py

```diff
@@ -0,0 +1,257 @@
+"""URL shortening module.
+
+The module is for all endpoints that manages all services related
+to url shortening.
+"""
+
+import dotenv
+from flask import request, abort, jsonify, g, redirect
+from werkzeug.exceptions import BadRequest
+from voluptuous import MultipleInvalid
+
+from app import db
+from app.api import api
+from app.api.validators import valid_url
+from app.api.auth import auth
+from app.helper import UrlSaver
+from app.models import LongUrl, ShortUrl, User
+
+
+dotenv.load()
+site_url = dotenv.get('SITE_URL')
+
+
+def check_authentication_with_token():
+    """Check that a user is authenticated and used token for authentication."""
+    if g.current_user.is_anonymous or not g.token_used:
```

Comment:
No. `g.token_used` only checks if the user used a token for authentication or not. The user could possibly have used an email and password combination for authentication.